### PR TITLE
nn_backup_restore

### PR DESCRIPTION
**Pull request by dmitrypekar**
Hi Elizabeth,

Please review&merge the initial approach to NN backup.
This covers basic scenario, that we've negotiated.

After this merge I will start working on JN backup&restore.

### CODE REVIEW

**Comment by elingg** [comment]
Thanks for your contribution, @dmitrypekar! I will do a detailed review. First, please make sure the build passes. We'll need some unit tests, but we can add them now or after the review as you see fit.

**Comment by dmitrypekar** [comment]
Hi Elizabeth, 
I've fixed all pmd, checkstyle and findbugs issue, that caused errors in build. You can proceed with the review now.

**Comment by elingg** [comment]
hey @dmitrypekar, sorry for the delay, but we're doing a refactor of the state store and the executors, see, https://github.com/mesosphere/hdfs/pull/216. In the interest of clean code, I would like that to go in first before we take a look at this PR,  (cc @gabrielhartmann @kensipe). Appreciate your patience here.

**Comment by dmitrypekar** [comment]
Ok, in that case I will work in parallel on the next hdfs item "Document
the disaster recovery procedure".

**Comment by elingg** [comment]
The other clean up changes have merged. Now would be the appropriate time to update this PR based on the refactoring. Thanks again for your patience, @dmitrypekar.

File: hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java [hunk 1]
```diff
@@ -148,6 +170,31 @@ public void frameworkMessage(ExecutorDriver driver, byte[] msg) {
     }
   }
 
+  private String getNodeId() {
```
Comment (elingg) [inline hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java]: I don't think we need to complex logic for the name of the backup directory. Would something simpler suffice?

File: hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsNode.java [hunk 2]
```diff
@@ -126,9 +118,9 @@ private ExecutorInfo createExecutor(String taskIdName, String nodeName, String e
               CommandInfo.URI
                 .newBuilder()
                 .setValue(
-                  String.format("http://%s:%d/%s", config.getFrameworkHostAddress(),
+                  String.format("http://%s:%d/%s?node=%s", config.getFrameworkHostAddress(),
```
Comment (elingg) [inline hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsNode.java]: Is it necessary to host the nodeId? Can we instead get this information from SchedulerConf?

**Comment by elingg** [comment]
Closing this as requested by @dmitrypekar

**Pull request closed without merging**